src/ui/views/Approval/components/SignTx.tsx

```diff
@@ -991,20 +1216,13 @@ const SignTx = ({ params, origin }: SignTxProps) => {
               noUpdate={isCancel || isSpeedUp}
               gasList={gasList}
               selectedGas={selectedGas}
+              version={txDetail.pre_exec_version}
               gas={{
-                ...(txDetail
-                  ? txDetail.gas
-                  : {
-                      estimated_gas_cost_usd_value: 0,
-                      estimated_gas_cost_value: 0,
-                      estimated_seconds: 0,
-                      estimated_gas_used: 0,
-                    }),
-                front_tx_count: 0,
-                max_gas_cost_usd_value: 0,
-                max_gas_cost_value: 0,
+                error: txDetail.gas.error,
+                estimated_gas_cost_usd_value: gasExplainResponse.gasCostUsd,
+                estimated_gas_cost_value: gasExplainResponse.gasCostAmount,
```

Comment:
success property not pass to GasSelecter means https://github.com/RabbyHub/Rabby/pull/909/files#diff-a3145b426c4c2574208bd357e4453077fee9e1e413c0bf688e074b58d8ae68afR318 condition will always be true

Reply:
![Uploading image.png…]()